src/graph/UserExecutor.cpp

```diff
@@ -0,0 +1,321 @@
+/* Copyright (c) 2019 vesoft inc. All rights reserved.
+ *
+ * This source code is licensed under Apache 2.0 License,
+ * attached with Common Clause Condition 1.0, found in the LICENSES directory.
+ */
+
+#include "graph/UserExecutor.h"
+#include "base/MD5Utils.h"
+
+namespace nebula {
+namespace graph {
+
+CreateUserExecutor::CreateUserExecutor(Sentence *sentence,
+                                       ExecutionContext *ectx) : Executor(ectx) {
+    sentence_ = static_cast<CreateUserSentence*>(sentence);
+}
+
+Status CreateUserExecutor::prepare() {
+    password_ = sentence_->getPassword();
+    missingOk_ = sentence_->getMissingOk();
+    userItem_.set_account(sentence_->getAccount()->data());
+    auto opts = sentence_->getOpts();
+    for (auto &opt : opts) {
+        switch (opt->getOptType()) {
+            case WithUserOptItem::OptionType::LOCK:
+            {
+                userItem_.set_is_lock(WithUserOptItem::asBool(opt->getOptVal()));
+                break;
+            }
+            case WithUserOptItem::OptionType::MAX_QUERIES_PER_HOUR:
+            {
+                userItem_.set_max_queries_per_hour(WithUserOptItem::asInt(opt->getOptVal()));
+                break;
+            }
+            case WithUserOptItem::OptionType::MAX_UPDATES_PER_HOUR:
+            {
+                userItem_.set_max_updates_per_hour(WithUserOptItem::asInt(opt->getOptVal()));
+                break;
+            }
+            case WithUserOptItem::OptionType::MAX_CONNECTIONS_PER_HOUR:
+            {
+                userItem_.set_max_connections_per_hour(WithUserOptItem::asInt(opt->getOptVal()));
+                break;
+            }
+            case WithUserOptItem::OptionType::MAX_USER_CONNECTIONS:
+            {
+                userItem_.set_max_user_connections(WithUserOptItem::asInt(opt->getOptVal()));
+                break;
+            }
+        }
+    }
+    return Status::OK();
+}
+
+void CreateUserExecutor::execute() {
+    auto *mc = ectx()->getMetaClient();
+    auto future = mc->createUser(userItem_, MD5Utils::md5Encode(*password_), missingOk_);
```

Comment:
maybe should move get the md5sum of password to nebula terminal is better. The password saved in memory directly is a high safe problem.

Reply:
@wadeliuyi  I agree with you, but now the `nebula terminal` does not parse the statement, so it can't know which is password

Reply:
Current processing logic：1. Save the encrypted password string to meta server when create user or change password. 2. encrypted the password string from client or terminal , and then compare password that have been stored. the connection can be made if the validation is successful, otherwise the connection is rejected